src/frequenz/sdk/actor/_decorator.py

```diff
@@ -201,25 +201,23 @@ async def _start_actor(self) -> None:
             """
             logger.debug("Starting actor: %s", cls.__name__)
             number_of_restarts = 0
-            while True:
+            while (
+                self.restart_limit is None or number_of_restarts <= self.restart_limit
+            ):
                 try:
+                    if number_of_restarts > 0:
+                        logger.info("Restarting actor: %s", cls.__name__)
```

Comment:
I would move this outside of the `try:` as you should only catch errors produced by the actor here, not your own code.

Other than that, great! This looks much cleaner now! :+1: